JavaScript/chapter02/2.1 - Remove Dups/solution.js

```diff
@@ -0,0 +1,71 @@
+class Node {
+  constructor(data) {
+    this.data = data;
+    this.next = null;
+  }
+}
+
+class LinkedList {
+  constructor() {
+    this.head = null;
+  }
+
+  //Function to insert a node in the singlely linkedlist.
+  insert(data) {
+    var newNode = new Node(data);
+    if (this.head == null) {
+      this.head = newNode;
+    } else {
+      var currentNode = this.head;
+      while (currentNode.next) {
+        currentNode = currentNode.next;
+      }
+
+      currentNode.next = newNode;
+    }
+  }
+
+  //Function to print the singlely linkedlist.
+  printList() {
+    let currentNode = this.head;
+    let list = "";
+    while (currentNode) {
+      list += currentNode.data + "->";
+      currentNode = currentNode.next;
+    }
+    list += "null";
+    console.log(list);
+  }
+
+  //Function to remove dupliicate nodes from the singlely linkedlist.
+  removeDuplicates() {
+    let currentNode1;
+    currentNode1 = this.head;
+    while (currentNode1 && currentNode1.next) {
+      let currentNode2 = currentNode1;
+      while (currentNode2.next) {
+        if (currentNode1.data == currentNode2.next.data) {
+          currentNode2.next = currentNode2.next.next;
+        } else {
+          currentNode2 = currentNode2.next;
+        }
+      }
+      currentNode1 = currentNode1.next;
+    }
+  }
+}
+//main function.
+var list1 = new LinkedList();
```

Comment:
I'm trying to make it easier to know if your code is correct, so I added support for [mocha](https://mochajs.org/) in https://github.com/techqueria/data-structures-and-algorithms/pull/47

Can you change the code so that it follows the pattern found in https://github.com/techqueria/data-structures-and-algorithms/blob/aa70628efaf99a825115cea3c1008a4ef4276816/JavaScript/chapter01/1.1%20-%20Is%20Unique/solution.js ? you might need to change `printList()` in favor of [`toString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString) so that you can use `assert.equal(list.toString(), '...');`

Reply:
Sure I will make the changes soon

Reply:
Maybe by using recursion time complexity can be reduce.

Reply:
that change has been merged now!